btfgen.c

```diff
@@ -113,9 +113,11 @@ int generate_btf(const char *src_btf, const char *dst_btf, const char *objspaths
 
 	err = btf__save_to_file(btf_new, dst_btf);
 	if (err) {
+		btf__free(btf_new);
 		printf("error saving btf file\n");
 		goto out;
 	}
+	btf__free(btf_new);
```

Comment:
What about moving it under the `out` label? something like:

```
if (!libbpf_get_error(btf_new))
    btf_free(btf_new);
```

In that way we'll have a common place to release it.

Reply:
it makes sense, I'll do that :)